sql/src/test/java/io/crate/testing/DataTypeTesting.java

```diff
@@ -109,11 +110,22 @@
                 return () -> (T) GeoShapeType.INSTANCE.value("POINT (10.2 32.2)");
 
             case ObjectType.ID:
-                Supplier<?> innerValueGenerator = getDataGenerator(randomType());
+                DataType<?> dataType = randomType();
                 return () -> {
                     // Can't use immutable Collections.singletonMap; insert-analyzer mutates the map
                     HashMap<String, Object> map = new HashMap<>();
-                    map.put("x", innerValueGenerator.get());
+
+                    // If GeoShapeType is chosen create a mutable Map otherwise ValueNormalizer
+                    // will fail when trying to set a value on an immutable Map.Entry
+                    if (dataType.id() == GeoShapeType.ID) {
```

Comment:
Couldn't we just change the GeoShapeType case ? (https://github.com/crate/crate/pull/6307/files#diff-82ef1bc82c97dc8e42f88544ac879527R110)

Reply:
Of course, I don't even know why I did that like this before :-)
Fixup squashed.